### PR TITLE
Sixth homework for check is ready ))

### PR DESCRIPTION
**Pull request by nvkaip**

### CODE REVIEW

File: Car.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public void setMaxSpeed(int maxSpeed) {
+        this.maxSpeed = maxSpeed;
+    }
+
+    public void setEngineValue(double engineValue) {
+        this.engineValue = engineValue;
+    }
+}
```
Comment (nvkaip) [inline Car.java]: Above is a class Car with overridden clone method to make it deep clone.
Next is a class for test clone method in Car class

File: CarCloneApp.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    public static void main(String[] args){
+        Car car = new Car("Audi", "A3", 200, 1.8);
+        Car carCloned = (Car) car.clone();
+        carCloned.setModel("A8");
+        car.setBrand("BMW");
+        car.setModel("X5");
+        System.out.println("Original " + car);
+        System.out.println("Cloned " + carCloned);
+    }
+}
```
Comment (nvkaip) [inline CarCloneApp.java]: Next is a Robot task with "try with resources". Main file is the last one in this list.

File: FuelType.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.homework;
+
+public enum FuelType{
+    PATROL,
+    DIESEL,
+    LPG
+}
```
Comment (nvkaip) [inline FuelType.java]: Next is a Robot task with "try with resources". Main file is the last one in this list.

File: Car.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    public Car deepClone(Car otherCar) {
+        Car cloned = new Car(otherCar.brand, otherCar.model, otherCar.engineType);
+        cloned.setEngineType(new EngineType(getEngineType().getFuelType(),
+                getEngineType().getEngineValue(),
+                getEngineType().getMaxSpeed()));
+        return cloned;
+    }
+
+    @Override
+    public Object clone() {
```
Comment (Sofasmile) [inline Car.java]: please use one of two way of making clone (override clone or copy constructor). If you use override clone then you need override this method in all classes.
you can see the example of implementation:
https://www.baeldung.com/java-deep-copy
https://www.javadevjournal.com/java/java-deep-copy/

File: hw06/Car.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        Car cloned;
+        try {
+            cloned = (Car) super.clone();
+        } catch (CloneNotSupportedException e) {
+            cloned = new Car(this.brand, this.model, this.engineType);
+        }
+        cloned.engineType = (EngineType) this.engineType.clone();
+        return cloned;
+    }
+
```
Comment (nvkaip) [inline hw06/Car.java]: Fixed clone method with clone EngineType

**Pull request merged**